lib/srv/desktop/windows_server.go

```diff
@@ -1330,3 +1335,59 @@ func (s *WindowsService) runCRLUpdateLoop(tlsConfig *tls.Config) {
 		}
 	}
 }
+
+// getKDCAddress performs DNS SRV lookups on the configured AD `domain`
+// to find KDC servers automatically. If `locate_server` is not set,
+// the `kdc_address` field is returned.
+func (s *WindowsService) getKDCAddress(ctx context.Context) (string, error) {
+	if !s.cfg.LocateServer.Enabled {
+		return s.cfg.KDCAddr, nil
```

Comment:
Should we also check LDAP addr here?

```suggestion
		return cmp.Or(s.cfg.KDCAddr, s.cfg.LDAP.Addr), nil
```